### ISSUE TITLE
[spec] size parameters description is not clear

### ISSUE THREAD

**Issue by khaledhosny** [adobe-type-tools/afdko#112]
The description of the size feature parameters is not very clear, namely:

> These values may also be specified more directly as decimal point values

It is not clear which values can be decimal point, all the four or only the decipoints ones? (my guess which makes since, but it was not clear the first, and even the second times I read the description).

> but a decimal point and following value is then required.

This not clear either, does this mean `parameters 10.0 0;` is accepted, but `parameters 10;` is not? If so, why?

Also it is not explicitly mentioned which value can be omitted and what the default will be.

(for the background I’m adding size feature support to FontTools, behdad/fonttools#550, and had to go to many iterations and still not sure my implementation is right).

**Comment by readroberts** [comment]
I agree that the wording is not clear.

The idea is that any of the values for the ‘parameter’ can be specified as an integer or as a real number. If it is specified as an integer, the value is interpreted as a decipoint. If it is specified as a real number, it is interpreted as a real number. “15” and “1.5” are interpreted as indicating the same value.

The phrase "but a decimal point and following value is then required” is trying to make the point that number string cannot end with a decimal point: an additional digit must follow the decimal point. For example,  “1.0" is valid input, but “1.” is not.

- Read

**Comment by readroberts** [comment]
PS. I see I did not comment on the default values. The answer is that there are no default values. If the font is part of a size menu group, all the fields must be filled out. If the font is not part of a size menu group, you can still use the size parameter to indicate the intended optical size for for the font. For this case, the first 'parameters' value is filled with a value indicating the intended optical size, and the rest of the rest of the parameters values are set to zero.

**Comment by khaledhosny** [comment]
Thanks for explanations. Still a few points are not clear:
* Does subfamily identifier accept floating point numbers?
* Source Sans Pro has the following size feature:
````
feature size {
        parameters 10.0 0; # design size, subfamily
} size;
````
What should be the value of the omitted fields, and can any field be omitted?

**Comment by readroberts** [comment]
The spec is clear: no values can be omitted. The 'size' feature definition from Source Sans Pro works because the makeotf logic differs from the spec. In makeotf, if the subfamily id is 0, then the other design values are also set to zero, whether or not there are more values in the list. If the subfamily ID is supplied and is not zero, and there are fewer than 4 values, then the the missing parameter values are set to whatever garbage is in memory.

I think I prefer the makeotf logic (except for the bug) as it seems odd to make the user specify 3 zeroes instead of 1. This alternate form also makes it more obvious that the definition is not for  use with a size menu group. Anyone object if I change the feature file spec to match?

**Comment by khaledhosny** [comment]
I’m fine either way.

**Comment by readroberts** [comment]
Fixed in feature file spec v 1.19
